tests/python/gaia-ui-tests/gaiatest/tests/functional/browser/test_browser_cards_in_cards_views.py

```diff
@@ -0,0 +1,55 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By, Wait
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.search.app import Search
+from gaiatest.apps.system.regions.cards_view import CardsView
+
+
+class TestBrowserCardsInCardsView(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+        self.test_url = self.marionette.absolute_url('mozilla.html')
+        self.test_url2 = self.marionette.absolute_url('mozilla_community.html')
+
+    def test_browser_cards_in_cards_views(self):
+        search = Search(self.marionette)
+        search.launch()
+        browser = search.go_to_url(self.test_url)
+        browser.wait_for_page_to_load()
+
+        browser.tap_menu_button()
+        search2 = browser.open_new_window()
+        search2.wait_to_be_displayed()
+        browser2 = search2.go_to_url(self.test_url2)
+        browser2.wait_for_page_to_load()
+        browser2.switch_to_content()
+        self.assertEqual('Mozilla Community', self.marionette.title)
+        browser2.switch_to_chrome()
+
+        cards_view = CardsView(self.marionette)
+        self.device.hold_home_button()
+        cards_view.wait_for_cards_view()
+        cards_view.cards[1].wait_for_centered()
+        self.assertEqual(len(cards_view.cards), 2, 'Should have 1 card to display')
+
+        self.assertEqual('Mozilla', cards_view.cards[0].title)
+        self.assertIn(cards_view.cards[0].subtitle, self.test_url)
+        self.assertEqual('Mozilla Community', cards_view.cards[1].title)
+        self.assertIn(cards_view.cards[1].subtitle, self.test_url2)
+
+        cards_view.swipe_to_previous_app()
```

Comment:
I wonder if we could make these 3 calls into 1 function:
```py
cards_view.open_app(position=0)
```
```py
def open_app(expected_position):
    # When you open the card views, you're always at the end of the list
    current_position = len(cards_view.cards)

    if expected_position >  current_position or expected_position < 0:
        raise IndexError

    while current_position > expected_position:
        self.swipe_to_previous_app()
        cards_view.cards[current_position].wait_for_centered()

    cards_view.cards[expected_position].tap()
```

What do you think?